### DISCUSSION TITLE
能否支持切面二开

### DISCUSSION THREAD

**Discussion by lpf4254302** [apache/iotdb#5554] [Ideas]
支持开发切面类，代理到查询、创建时序、创建存储组等功能
例如：
- 用户希望在查询后筛选出查询时间超过3秒的所有sql，并报警或插入数据库等后续操作；
- 可能安全部门有需求，要求从数据库层面监控敏感词，那么需要再创建操作/插入操作时，对字符进行监控，并可能进行报警；

PS：需求方也可以自己利用动态代理实现，如果原生支持会更好些

**Comment by jixuan1989** [comment]
Maybe we could implement some annotation to enable this feature if needed.
